packages/demo/src/app/connected/(tools)/TimeLockedTransfer/page.tsx

```diff
@@ -0,0 +1,224 @@
+  const { log, error } = createSender("Transfer");
+
+  const { explorerTransaction } = useGetExplorerLink();
+
+  const [transferTo, setTransferTo] = useState<string>("");
+  const [amount, setAmount] = useState<string>("");
+  const [data, setData] = useState<string>("");
+  const [lockedForBlocks, setLockedForBlocks] = useState<string>("");
+  const [liveTimeLockCell, setLiveTimeLockCell] = useState<ccc.Cell | null>(null);
+  const [showClaimPage, setShowClaimPage] = useState<boolean>(false);
+  const [claimData, setClaimData] = useState<string>("");
+
+  const handleTimeLockedTransfer = async () => {
+    if (!signer) {
+      return;
+    }
+
+    // Verify destination addresses
+    const toAddress = await ccc.Address.fromString(transferTo, signer.client);
+
+    const currentBlockNumber = await signer.client.getTip();
+    const lockedUntil = new Since("absolute", "blockNumber", currentBlockNumber + numFrom(lockedForBlocks));
+    const lockedUntilToNum = lockedUntil.toNum();
+    const timeLockScript = ccc.Script.from({
+      ...testnetTimeLockScript,
+      args: buildTimeLockArgs(toAddress.script.hash(), lockedUntilToNum.toString()),
+    })
+
+    const timeLockCellOutput = { lock: timeLockScript };
+
+    const tx = ccc.Transaction.from({
+      outputs: [timeLockCellOutput],
+      outputsData: [bytesFromAnyString(data)],
+    });
+
+    const minimumCapacity = tx.getOutputsCapacity();
+    if (minimumCapacity > ccc.fixedPointFrom(amount)) {
+      error(`Insufficient capacity to store data`);
+      return;
+    }
+    tx.outputs[0].capacity = ccc.fixedPointFrom(amount);
+
+    // Complete missing parts for transaction
+    await tx.completeInputsByCapacity(signer);
+    await tx.completeFeeBy(signer, feeRate);
+
+    signer.sendTransaction(tx).then((hash: string) => {
+      log(
+        "Transaction sent:",
+        explorerTransaction(hash),
+      );
+    }).catch((err: Error) => {
+      error(err.message);
+    });
+  };
+
+  useEffect(() => {
+    checkTimeLockedCells();
+  }, []);
+
+  const checkTimeLockedCells = async () => {
+    if (!signer) {
+      return;
+    }
+    // cell ordering is not guaranteed
+    const timeLockedCells: ccc.Cell[] = [];
+    for await (const cell of signer.client.findCells(
+      {
+        script: ccc.Script.from({
+          ...testnetTimeLockScript,
+          args: bytesFromAnyString((await signer.getRecommendedAddressObj()).script.hash()),
+        }),
+        scriptType: "lock",
+        scriptSearchMode: "prefix",
+      },
+      "desc",
+    )) {
+      // Process each cell
+      timeLockedCells.push(cell);
+    }
+    // pick the first one
+    setLiveTimeLockCell(timeLockedCells[0]);
+  }
+
+  const handleClaim = async () => {
+    if (!signer || !liveTimeLockCell) {
+      return;
+    }
+    const toAddress = await signer.getRecommendedAddressObj();
+    const tx = ccc.Transaction.from({
+      outputs: [{ lock: toAddress.script }],
+      inputs: [ccc.CellInput.from({
```

Comment:
It's not necessary to add a `CellInput.from` here. CCC helps you to do this automatically.

Reply:
I believe that explicitly passing the target input cell to be consumed is the safer choice here. For instance, in a claiming scenario, if a user intends to claim an input cell containing 1,000 CKB, they would expect to spend that exact input cell. Allowing CCC to automatically compose inputs could result in selecting two different input cells—for example, one with 500 CKB and another with 600 CKB. Moreover, these two inputs may not be readily available for the user to claim, whereas the original input cell is. The discrepancy in the number of input cells and their lock time could lead to overall confusion.

Reply:
Sorry for the misunderstanding...